readium/navigator/src/main/java/org/readium/r2/navigator/epub/EpubNavigatorFragment.kt

```diff
@@ -1035,21 +1037,27 @@ class EpubNavigatorFragment internal constructor(
          * if you use a local HTTP server.
          * @param initialLocator The first location which should be visible when rendering the
          * publication. Can be used to restore the last reading location.
+         * @param readingOrder custom reading order
          * @param listener Optional listener to implement to observe events, such as user taps.
          * @param config Additional configuration.
          */
         fun createFactory(
             publication: Publication,
             baseUrl: String? = null,
             initialLocator: Locator? = null,
+            readingOrder: List<Link>? = null,
             listener: Listener? = null,
             paginationListener: PaginationListener? = null,
             config: Configuration = Configuration(),
             initialPreferences: EpubPreferences = EpubPreferences()
         ): FragmentFactory =
             createFragmentFactory {
                 EpubNavigatorFragment(
-                    publication, baseUrl, initialLocator, initialPreferences,
+                    publication, baseUrl, initialLocator,
+                    readingOrder = readingOrder ?: publication.readingOrder,
+                    positionsByReadingOrder = if (readingOrder == null)
+                        Collections.emptyList() else runBlocking { publication.positionsByReadingOrder() },
```

Comment:
Do we really want to expose a `positionsByReadingOrder` argument? I feel like our approach is fine for now.

Reply:
Agree - this is more me not being 100% sure on how to solve the constructor arguments in Kotlin... Maybe one of you can suggest something here?

Reply:
Yes you could take only the `readingOrder: List<Link>?` in the internal constructor and initialize the internal properties. This way you won't have to do the same thing in `EpubNavigatorFactory` either.

```kotlin
private val readingOrder: List<Link> = readingOrder ?: publication.readingOrder

private val positionsByReadingOrder: List<List<Locator>> =
    if (readingOrder == null) emptyList()
    else runBlocking { publication.positionsByReadingOrder() }
```

Reply:
For `positionsByReadingOrder` to be able to test for null on readingOrder, I kept `readingOrder` as List<Link>? = null in the constructor.

Instead we check for null when we try to use `readingOrder` to not break `positionsByReadingOrder`.

Reply:
Wow, good catch, and sorry 🙀 - I was a bit quick here!! 

I'll see if I can find out about the constructor.

Reply:
But if I declare the property with the same name - how would positionsByReadingOrder be able to check if the constructor argument for readingOrder was null? @qnga

Reply:
Seems like that is automatically solved in the positionsByReadingOrder property - so then we should be all good. Pushed a few updates.